batch-index/src/test/java/no/unit/nva/search/EventBasedBatchIndexerTest.java

```diff
@@ -69,17 +72,22 @@ public void batchIndexerParsesEvent() {
     @ValueSource(ints = {1, 2, 5, 10, 50, 100})
     public void shouldIndexNFilesPerEvent(int numberOfFilesPerEvent) throws IOException {
         indexer = new EventBasedBatchIndexer(s3Client, elasticSearchClient, eventBridgeClient, numberOfFilesPerEvent);
-        var expectedFiles = randomFilesInSinceEvent(s3Driver, numberOfFilesPerEvent);
+        var expectedFiles = randomFilesInSingleEvent(s3Driver, numberOfFilesPerEvent);
         var unexpectedFile = randomEntryInS3(s3Driver);
 
         var importLocation = unexpectedFile.getHost().getUri(); //all files are in the same bucket
         InputStream event = eventStream(new ImportDataRequest(importLocation.toString()));
         indexer.handleRequest(event, outputStream, CONTEXT);
 
         for (var expectedFile : expectedFiles) {
-            assertThat(elasticSearchClient.getIndex(), hasKey(expectedFile.getFilename()));
+            IndexDocument indexDocument = fetchIndexDocumentFromS3(expectedFile);
```

Comment:
Each index document contains information about the index it is supposed to be indexed in. We have to read the file that we want to index to find out where to look in the Fake Search service